src/uu/uniq/src/uniq.rs

```diff
@@ -390,12 +431,30 @@ pub fn uu_app() -> Command {
                 .help("end lines with 0 byte, not newline")
                 .action(ArgAction::SetTrue),
         )
+        .group(
+            ArgGroup::new(options::OBSOLETE_SKIP_FIELDS)
+            .multiple(false)                       // throw an error on `uniq -3 -21` and `uniq -f 4 -3` rather than giving a wrong result
+            .conflicts_with(options::SKIP_FIELDS)  // for consistency with other uutils arguments, while in gnu
+            .args(OBSOLETE_SKIP_FIELDS_DIGITS)     // `uniq -3 -21` would be equal to `uniq -f 321`
+        )                                          // and `uniq -f 4 -3` would be equal to `uniq -f 3`
```

Comment:
I don't like this formatting with comments at the end. It's hard to see what they refer to (is each comment about the line they are on, are they all about `multiple`, are they all about the whole block?). Placing this all above the `ArgGroup` makes it much simpler. 

Also it needs a bit more explanation because people reading these comments won't have the full context of this whole PR. Here's my attempt:

> GNU `uniq` parses each digit of the shorthand as a short flag. For example, `uniq -21` is parsed as `uniq -2 -1` and the flags are then effectively concatenated so that `-2 -1` has the behaviour of `-f 21`. This is hard to do in clap and leads to some surprising outcomes (`uniq -2 foo -1` being equal to `uniq -21 foo`), so we parse the shorthand more like `fold`: we have a short flag for each digit, which takes the rest of the number as value.
> This means that `uniq -1 -2` would behave different from GNU, so we explicitly disallow using the shorthand multiple times.
> Additionally, these arguments conflict with `-f`/`--skip-fields`, because they have overriding behaviour in GNU that we don't support here.

Also, I _think_ we can actually support that case with `-f` by calling `.overrides_with(options::SKIP_FIELDS)` instead of `.conflicts_with(options::SKIP_FIELDS)`, but that does require some testing to be sure.

Reply:
thanks, initially i meant to use `overrides_with()`, but while it worked with `-N` followed by `-f N`, it gave errors, with `-f N` followed by `-N`, so i assumed it wasn't doable between a single option and a group  
but i just found out that on the other hand using `overrides_with_all(OBSOLETE_SKIP_FIELDS_DIGITS)` on SKIP_FIELDS works both ways